### PR TITLE
SED-2701 temporary-ap-resources-not-cleanup-properly-after-ap-update

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java [hunk 1]
```diff
@@ -318,11 +314,22 @@ protected Staging createStaging(){
...
                                          ObjectEnricher objectEnricher) {
+        List<Resource> stagingResources = staging.resourceManager.findManyByCriteria(null);
+        try {
+            for (Resource resource: stagingResources) {
+                resourceManager.copyResource(resource, staging.resourceManager);
+            }
+        } catch (IOException | SimilarResourceExistingException | InvalidResourceFormatException e) {
+            throw new AutomationPackageManagerException("Unable to persist a resource in automation package", e);
+        } finally {
+            staging.resourceManager.cleanup();
```
Comment (jeromecomte) [inline step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java:330]: Not sure if it is the best place to cleanup the staging resource manager but it does the job for the moment

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java [hunk 2]
```diff
@@ -627,6 +635,7 @@ protected static class Staging {
         List<Plan> plans = new ArrayList<>();
         List<ExecutiontTaskParameters> taskParameters = new ArrayList<>();
         List<Function> functions = new ArrayList<>();
+        ResourceManager resourceManager = new LocalResourceManagerImpl(new File("resources", new ObjectId().toString()));
```
Comment (jeromecomte) [inline step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java:638]: I would maybe use a dedicated context name "ap_staging_resources" for instance in order to quickly identify the root cause of potential leaks in the future

**Review by jeromecomte — APPROVED** [review]
Approving with 2 minor remarks

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java (hunk 2 above)
Comment (david-stephan) [reply step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java]: renamed as suggested in last commit

**Pull request merged**